diff --git a/src/backend/imap/imap.php b/src/backend/imap/imap.php
index cffe62e4..232d1f4f 100644
--- a/src/backend/imap/imap.php
+++ b/src/backend/imap/imap.php
@@ -1777,10 +1777,6 @@ class BackendIMAP extends BackendDiff implements ISearchProvider {
 
         $imapid = $this->getImapIdFromFolderId($folderid);
         $uid = $this->getUidFromId($folderid, $id);
-        if (strcasecmp($imapid, $this->create_name_folder(IMAP_FOLDER_TRASH)) != 0) {
-            ZLog::Write(LOGLEVEL_DEBUG, sprintf("BackendIMAP->DeleteMessage('%s','%s') move message to trash folder", $folderid, $id));
-            return $this->MoveMessage($folderid, $id, $this->create_name_folder(IMAP_FOLDER_TRASH), $contentparameters);
-        }
         $this->imap_reopen_folder($imapid);
 
         if ($this->imap_inside_cutoffdate(Utils::GetCutOffDate($contentparameters->GetFilterType()), $uid)) {
